libcore/gof-directory-async.vala

```diff
@@ -235,6 +249,30 @@ public class Async : Object {
         yield make_ready (is_no_info || success, file_loaded_func); /* Only place that should call this function */
     }
 
+    public void update_git_status () {
+        if (git_repo == null) {
+            return;
+        }
+
+        var options = new Ggit.StatusOptions (Ggit.StatusOption.INCLUDE_UNTRACKED, Ggit.StatusShow.INDEX_AND_WORKDIR, null);
+
+        try {
+            /* FIXME This seg faults for unknown reasons */
```

Comment:
Have you tried changing the `check_each_git_status` function from private to public?

Reply:
Thanks Vishal - David Hewitt advised that the problem is that I had not initialised the Ggit library before using this function.  Confusingly, all the other Ggit library functions that I used worked OK so this was not easy to spot.